File: backend/app/schemas/action_item.py
from datetime import date, datetime
from enum import Enum

from pydantic import BaseModel, Field


class ActionStatus(str, Enum):
    TODO = "todo"
    DOING = "doing"
    DONE = "done"
    CANCELED = "canceled"


class Priority(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    URGENT = "urgent"


class ActionItemBase(BaseModel):
    task: str = Field(min_length=1)
    owner: str = Field(min_length=1, max_length=100)
    deadline: date | None = None
    priority: Priority = Priority.MEDIUM
    source_chunk_id: str | None = None


class ActionItemCreate(ActionItemBase):
    pass


class ActionItemUpdate(BaseModel):
    task: str | None = Field(default=None, min_length=1)
    owner: str | None = Field(default=None, min_length=1, max_length=100)
    deadline: date | None = None
    priority: Priority | None = None
    status: ActionStatus | None = None


class ActionItemRead(ActionItemBase):
    id: str
    meeting_id: str
    status: ActionStatus
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}

File: frontend/meetingPilot-frontend/src/types/action_item.ts
export enum ActionStatus {
  TODO = 'todo',
  DOING = 'doing',
  DONE = 'done',
  CANCELED = 'canceled',
}

export enum Priority {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
  URGENT = 'urgent',
}

export interface ActionItemBase {
  task: string
  owner: string
  deadline: string | null
  priority: Priority
  source_chunk_id: string | null
}

export interface ActionItemCreate extends ActionItemBase {}

export interface ActionItemUpdate {
  task?: string | null
  owner?: string | null
  deadline?: string | null
  priority?: Priority | null
  status?: ActionStatus | null
}

export interface ActionItemRead extends ActionItemBase {
  id: string
  meeting_id: string
  status: ActionStatus
  updated_at: string | null
}

export interface ActionListResponse {
  meeting_id: string
  items: ActionItemRead[]
}

export interface FeishuSyncResponse {
  action_id: string
  provider: string
  external_id: string
  status: string
  synced_at: string
}
